lib/stitches/utils/autoroute.py

```diff
@@ -83,39 +83,71 @@ def add_jumps(graph, elements, preserve_order):
     Jump stitches are added to ensure that all elements can be reached.  Only the
     minimal number and length of jumps necessary will be added.
     """
-
     if preserve_order:
-        # For each sequential pair of elements, find the shortest possible jump
-        # stitch between them and add it.  The directions of these new edges
-        # will enforce stitching the elements in order.
+        graph = _add_ordered_jumps(graph, elements)
+    else:
+        graph = _add_unordered_jumps(graph, elements)
+    return graph
 
-        for element1, element2 in zip(elements[:-1], elements[1:]):
-            check_stop_flag()
 
-            potential_edges = []
+def _add_ordered_jumps(graph, elements):
+    # For each sequential pair of elements, find the shortest possible jump
+    # stitch between them and add it.  The directions of these new edges
+    # will enforce stitching the elements in order.
+    for element1, element2 in zip(elements[:-1], elements[1:]):
+        check_stop_flag()
+        graph = _insert_smallest_jump(graph, element1, element2)
```

Comment:
I don't think we need to return `graph` from each of these functions like this.  We're just adding nodes and edges, not creating entirely new graphs.

Reply:
Ok, removed it.